wp-parsely.php

```diff
@@ -46,3 +48,48 @@ function wpvip_load_wp_parsely_plugin() {
 	require 'wp-parsely-' . $major_version . '/wp-parsely.php';
 }
 add_action( 'after_setup_theme', 'wpvip_load_wp_parsely_plugin' );
+
+/**
+ * Pre-sets the "apikey" portion of the `parsely` site option hash map to the site_url IF it's empty.
+ * This is only hooked into `option_parsely` if we're loading the plugin from this repo.
+ *
+ * @param array $parsely_settings Settings from the database
+ * @return array Settings with the 'apikey' value overridden
+ */
+function wpvip_override_parsely_option_if_empty( $parsely_settings ) {
```

Comment:
```suggestion
function wpvip_override_parsely_option_if_empty( array $parsely_settings ): array {
```

Reply:
Thanks for the review, @pauarge!

Unless I'm missing something, I don't think we can do the `array` type declaration on the argument as an uninitialized option will result in an uncaught `TypeError` on newer PHP versions:

e.g.

```
# This is in PHP 7.4.21
php -r 'function func_that_takes_an_array( array $maybe_an_array = [] ) {} func_that_takes_an_array( false );'
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to func_that_takes_an_array() must be of the type array, bool given, called in Command line code on line 1 and defined in Command line code:1
```

I think the return type is safe, however.

Reply:
In which case, as [per WP core](https://github.com/WordPress/WordPress/blob/39cb388519cc573b9fae6db2c3fe94a66bee0d41/wp-includes/option.php#L221), the `@param` should be `mixed`, not `array` on line 56.

Reply:
If the input type can be `mixed`, should we update the type on the doc comment above?

Reply:
Yes - that's line 56 :-)